apps/hash-frontend/src/pages/notes.page/convert-quick-note-to-page-modal.tsx

```diff
@@ -0,0 +1,250 @@
+import {
+  Autocomplete,
+  IconButton,
+  Modal,
+  TextField,
+} from "@hashintel/design-system";
+import { types } from "@local/hash-isomorphic-utils/ontology-types";
+import { Entity, OwnedById } from "@local/hash-subgraph/.";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+import {
+  autocompleteClasses,
+  Box,
+  Divider,
+  ModalProps,
+  outlinedInputClasses,
+  Typography,
+} from "@mui/material";
+import { generateKeyBetween } from "fractional-indexing";
+import { FunctionComponent, useCallback } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import { useBlockProtocolCreateEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-create-entity";
+import { useBlockProtocolUpdateEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
+import { useAccountPages } from "../../components/hooks/use-account-pages";
+import { PageIcon } from "../../components/page-icon";
+import { GetAccountPagesTreeQuery } from "../../graphql/api-types.gen";
+import { XMarkRegularIcon } from "../../shared/icons/x-mark-regular-icon";
+import { Button } from "../../shared/ui";
+import { useAuthenticatedUser } from "../shared/auth-info-context";
+import { QuickNoteEntityWithCreatedAt } from "./types";
+
+type Page = GetAccountPagesTreeQuery["pages"][number];
+
+export type PageWithParentLink = Page & { parentLinkEntity?: Entity };
+
+type ConvertToPageFormData = {
+  title?: string;
+  parentPage?: Page;
+};
+
+export const ConvertQuickNoteToPageModal: FunctionComponent<
+  Omit<ModalProps, "children" | "onClose"> & {
+    onClose: () => void;
+    onConvertedToPage: (page: PageWithParentLink) => void;
+    quickNoteEntityWithCreatedAt: QuickNoteEntityWithCreatedAt;
+  }
+> = ({
+  quickNoteEntityWithCreatedAt,
+  onClose,
+  onConvertedToPage,
+  ...modalProps
+}) => {
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const { control, reset, handleSubmit, register, setValue } =
+    useForm<ConvertToPageFormData>();
+
+  const defaultTitle = `Quick Note - ${quickNoteEntityWithCreatedAt.createdAt.toLocaleString()}`;
+
+  const { data: pages } = useAccountPages(
+    authenticatedUser.accountId as OwnedById,
+  );
+
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+
+  const { createEntity } = useBlockProtocolCreateEntity(
+    authenticatedUser.accountId as OwnedById,
+  );
+
+  const innerSubmit = handleSubmit(async (data) => {
+    const { parentPage } = data;
+
+    const title = data.title ?? defaultTitle;
```

Comment:
The `defaultTitle` is not being used, because `data.title` defaults to an empty string

Reply:
Thanks for catching, have addressed in https://github.com/hashintel/hash/pull/3434/commits/0f11122e4bff38d8180c694be32d9414b1ba720e

Reply:
Oh right yeah, have fixed in https://github.com/hashintel/hash/pull/3434/commits/4dcf1557220a87a2371a3bde5eb0392a348e1874